### PR TITLE
adds new deps.

### PR DESCRIPTION
**Pull request by mfaghihi**

### CODE REVIEW

File: node/Cargo.toml [hunk 1]
```diff
@@ -93,9 +93,14 @@ polkadot-primitives = { git = "https://github.com/paritytech/polkadot", branch =
 polkadot-service = { git = "https://github.com/paritytech/polkadot", branch = "master" }
 polkadot-test-service = { git = "https://github.com/paritytech/polkadot", branch = "master" }
 
-arkworks-gadgets = { version = "^0.3", features = ["default_poseidon"] }
+arkworks-gadgets = { version = "0.4.1", features = ["std", "r1cs"], default-features = false }
```
Comment (drewstone) [inline node/Cargo.toml:96]: Worth bumping to 0.4.2

Comment (drewstone) [reply node/Cargo.toml:96]: Need to remove features as well.

File: pallets/anchor/Cargo.toml [hunk 2]
```diff
@@ -50,7 +50,9 @@ ark-serialize = { version = "^0.3.0", default-features = false, features = [ "de
 ark-std = { version = "^0.3.0", default-features = false }
 ark-bn254 = { version = "^0.3.0", default-features = false, features = [ "curve" ] }
 ark-bls12-381 = { version = "^0.3.0", default-features = false, features = [ "curve" ] }
-arkworks-gadgets = { version = "^0.3", features = ["default_poseidon"] }
+arkworks-gadgets = { version = "0.4.1", features = ["std", "r1cs"], default-features = false }
```
Comment (drewstone) [inline pallets/anchor/Cargo.toml:53]: Need to remove features.

File: pallets/anchor/Cargo.toml [hunk 3]
```diff
@@ -50,7 +50,9 @@ ark-serialize = { version = "^0.3.0", default-features = false, features = [ "de
...
+arkworks-circuits = { version = "0.4.1", features = ["std", "r1cs"], default-features = false }
```
Comment (drewstone) [inline pallets/anchor/Cargo.toml:54]: Need to remove features.

File: pallets/anchor/Cargo.toml [hunk 4]
```diff
@@ -50,7 +50,9 @@ ark-serialize = { version = "^0.3.0", default-features = false, features = [ "de
...
+arkworks-utils = { version = "0.4.1", features = ["std", "r1cs"], default-features = false }
```
Comment (drewstone) [inline pallets/anchor/Cargo.toml:55]: Need to remove features.

File: node/Cargo.toml (hunk 1 above)
Comment (drewstone) [reply node/Cargo.toml:96]: Need to remove features.

File: pallets/anchor/zk-setup/Cargo.toml [hunk 5]
```diff
@@ -13,6 +13,7 @@ ark-serialize = { version = "^0.3.0", default-features = false, features = [ "de
 ark-ff = { version = "^0.3.0", default-features = false }
 ark-std = { version = "^0.3.0", default-features = false }
 ark-bn254 = { version = "^0.3.0", default-features = false, features = [ "curve" ] }
-arkworks-gadgets = { version = "^0.3", features = ["default_poseidon"] }
-
+arkworks-gadgets = { version = "0.4.1", features = ["std", "r1cs"], default-features = false }
```
Comment (drewstone) [inline pallets/anchor/zk-setup/Cargo.toml:16]: Need to remove features. And in the rest of the deps.

**Review by drewstone — CHANGES_REQUESTED** [review]
The runtime does not support the std library. Therefore you can't add these types of features here. You have to ensure there are no std dependencies in these crates or else everything will break.

**Comment by drewstone** [comment]
Is this ready for merge @mfaghihi ? We'll need a Pr updating the branches to versions also once this is merged since it points to a git branch.

**Review by filiplazovic — APPROVED** [review]

**Pull request merged**